test/e2e_node/topology_manager_test.go

```diff
@@ -862,6 +906,153 @@ func runTopologyManagerNodeAlignmentSuiteTests(ctx context.Context, f *framework
 	}
 }
 
+func runPreferClosestNUMATestSuite(ctx context.Context, f *framework.Framework, numaNodes int, distances map[int][]int) {
+	runPreferClosestNUMAOptimalAllocationTest(ctx, f, numaNodes, distances)
+	runPreferClosestNUMASubOptimalAllocationTest(ctx, f, numaNodes, distances)
+}
+
+func runPreferClosestNUMAOptimalAllocationTest(ctx context.Context, f *framework.Framework, numaNodes int, distances map[int][]int) {
+	ginkgo.By("Admit two guaranteed pods. Both consist of 1 containers, each pod asks for cpus from 2 NUMA nodes. CPUs should be assigned from closest NUMA")
+	podMap := make(map[string]*v1.Pod)
+	for podID := 0; podID < 2; podID++ {
+		numCores := 0
+		for nodeNum := 0 + 2*podID; nodeNum <= 1+2*podID; nodeNum++ {
+			cpus, err := getCPUsPerNUMANode(nodeNum)
+			framework.ExpectNoError(err)
+			// subtract one to accommodate reservedCPUs. It'll only work if more than 2 cpus per NUMA node.
+			cpusPerNUMA := len(cpus)
+			if cpusPerNUMA < 3 {
+				e2eskipper.Skipf("Less than 3 cpus per NUMA node on this system. Skipping test.")
+			}
+			numCores += cpusPerNUMA - 1
+		}
+		coresReq := fmt.Sprintf("%dm", numCores*1000)
+		ctnAttrs := []tmCtnAttribute{
+			{
+				ctnName:    "ps-container-0",
+				cpuRequest: coresReq,
+				cpuLimit:   coresReq,
+			},
+		}
+		podName := fmt.Sprintf("gu-pod-%d", podID)
+		framework.Logf("creating pod %s attrs %v", podName, nil)
```

Comment:
there are more instances of logging hardcoded nil, but for the sake of time I don't think this is worth blocking the PR. I'd rather spend time thinking about more meaningful tests and add them